mlflow/xgboost/__init__.py

```diff
@@ -361,8 +362,34 @@ def predict(
         Returns:
             Model predictions.
         """
+        import xgboost as xgb
+
         predict_fn = _wrapped_xgboost_model_predict_fn(self.xgb_model)
-        return predict_fn(dataframe, **(params or {}))
+        params = params or {}
+        # filter is applied inside predict_fn wrapper for xgb.Booster
+        if not isinstance(self.xgb_model, xgb.Booster):
+            params = _exclude_unrecognized_kwargs(predict_fn, params)
```

Comment:
can we also add a unit test to cover this case?

Reply:
and I suggest to emit a warning in this case

Reply:
I added test for Booster, what warnings do you want to emit? _exclude_unrecognized_kwargs internally has warnings

Reply:
oh yes, warning has been added in _exclude_unrecognized_kwargs . thanks!